pkg/controllers/localbuild/argo.go

```diff
@@ -53,7 +68,74 @@ func (r *LocalbuildReconciler) ReconcileArgo(ctx context.Context, req ctrl.Reque
 	if result, err := argocd.Install(ctx, resource, r.Client, r.Scheme, r.Config); err != nil {
 		return result, err
 	}
-
 	resource.Status.ArgoCD.Available = true
+
+	// Let's patch the existing argocd admin secret if devmode is enabled to set the default password
+	if r.Config.DevMode {
+		// Hash password using bcrypt
+		password := argocdDevModePassword
+		hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), 0)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("Error hashing password: %w", err)
+		}
+
+		// Prepare the patch for the Secret's `stringData` field
+		patchData := map[string]interface{}{
+			"stringData": map[string]string{
+				"accounts.developer.password":      string(hashedPassword),
+				"accounts.developer.passwordMtime": time.Now().Format(time.RFC3339),
+			},
+		}
+		// Convert patch data to JSON
+		patchBytes, err := json.Marshal(patchData)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("Error marshalling patch data: %w", err)
+		}
+
+		kubeClient, err := k8s.GetKubeClient()
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("getting kube client: %w", err)
+		}
+
+		// Getting the argocd-secret
+		s := v1.Secret{}
+		err = kubeClient.Get(ctx, client.ObjectKey{Name: argocdAdminSecretName, Namespace: argocdNamespace}, &s)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("getting argocd secret: %w", err)
+		}
+
+		// Patching the argocd-secret with the user's hashed password
+		err = kubeClient.Patch(ctx, &s, client.RawPatch(types.StrategicMergePatchType, patchBytes))
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("Error patching the Secret: %w", err)
+		}
```

Comment:
You can use server side apply instead here to simplify this process.

Reply:
Ok. I will review

Reply:
What do you mean by "server side" ? My code is similar to what we do here with the Gitea - https://github.com/cnoe-io/idpbuilder/blob/f31a08e5d8f023afaeca08a9db8e88884af755e1/pkg/controllers/localbuild/gitea.go#L189 ...

Reply:
The way it's one in Gitea is server side apply. If you look further up, you'll see it's working with unstructured object to avoid having ownership of fields we do not care about. Sometimes we see errors like "object has been updated and version doesn't match". Server side apply avoids that. 

This talks about it: https://eng.d2iq.com/blog/conflict-resolution-kubernetes-server-side-apply/

Here's where unstructured object is used:
https://github.com/cnoe-io/idpbuilder/blob/f31a08e5d8f023afaeca08a9db8e88884af755e1/pkg/controllers/localbuild/gitea.go#L163-L166

Here's another example: 
https://github.com/cnoe-io/idpbuilder/blob/d14289a40c6fed3909e1686e48d40acef4df8b8e/pkg/util/util.go#L78-L86

Reply:
Fixed. See: 80a785b